src/ApiCore/TransportFactory.php

```diff
@@ -35,99 +35,40 @@
-     *           Exactly one of $scopes or $credentialsLoader must be provided.
-     *     @type Channel $channel
-     *           Optional. A `Channel` object. If not specified, a channel will be constructed.
-     *           NOTE: This option is only valid when utilizing the gRPC transport.
-     *     @type ChannelCredentials $sslCreds
-     *           Optional. A `ChannelCredentials` object for use with an SSL-enabled channel.
-     *           Default: a credentials object returned from
-     *           \Grpc\ChannelCredentials::createSsl()
-     *           NOTE: This option is only valid when utilizing the gRPC transport. Also, if the $channel
-     *           optional argument is specified, then this argument is unused.
-     *     @type bool $forceNewChannel
-     *           Optional. If true, this forces gRPC to create a new channel instead of using a persistent channel.
-     *           Defaults to false.
-     *           NOTE: This option is only valid when utilizing the gRPC transport. Also, if the $channel
-     *           optional argument is specified, then this option is unused.
-     *     @type bool $enableCaching
-     *           Optional. Enable caching of access tokens. Defaults to true.
-     *     @type CacheItemPoolInterface $authCache
-     *           Optional. A cache for storing access tokens. Defaults to a simple in memory implementation.
-     *     @type array $authCacheOptions
-     *           Optional. Cache configuration options.
-     *     @type callable $authHttpHandler
-     *           A handler used to deliver PSR-7 requests specifically for
-     *           authentication. Should match a signature of
-     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
-     *     @type callable $httpHandler
-     *           A handler used to deliver PSR-7 requests. Should match a
-     *           signature of
-     *           `function (RequestInterface $request, array $options) : PromiseInterface`.
-     *           NOTE: This option is only valid when utilizing the REST transport.
+     * @param string $serviceAddress The address of the API remote host. Must be formatted as
+     *                               "<address>:port", e.g. "my.service.com:443"
+     * @param AuthWrapper $authWrapper AuthWrapper to manage auth tokens.
+     * @param array  $args {
      *     @type string $transport
-     *           The type of transport to build.
+     *           Optional. The type of transport to build. Defaults to
+     *           'grpc' when available. Supported values: ['grpc', 'rest'].
+     *     @type string $restClientConfigPath Path to rest client config.
+     *           Required for 'rest' transport only.
      * }
      * @return TransportInterface
+     * @throws ValidationException
      */
-    public static function build(array $args)
+    public static function build($serviceAddress, $authWrapper, array $args)
```

Comment:
The original intent is that users may choose to be using this factory to instantiate their transport, and this feels like a bit of a nerf. Prior to this change, if you wanted to use the factory to build your transport it was as straight forward as passing in the service address and port at minimum. Now you have to concern yourself with the concept of the auth wrapper, as well as how to construct one. I may be missing something here, I just don't see the argument for this change.